dictionaries/CallMap.php

```diff
@@ -15643,46 +15643,46 @@
 'xmlrpc_server_register_method' => ['bool', 'server'=>'resource', 'method_name'=>'string', 'function'=>'string'],
 'xmlrpc_set_type' => ['bool', '&rw_value'=>'string|DateTime', 'type'=>'string'],
 'XMLWriter::endAttribute' => ['bool'],
-'XMLWriter::endCData' => ['bool'],
+'XMLWriter::endCdata' => ['bool'],
 'XMLWriter::endComment' => ['bool'],
 'XMLWriter::endDocument' => ['bool'],
-'XMLWriter::endDTD' => ['bool', 'xmlwriter='=>''],
-'XMLWriter::endDTDAttlist' => ['bool'],
-'XMLWriter::endDTDElement' => ['bool'],
-'XMLWriter::endDTDEntity' => ['bool'],
+'XMLWriter::endDtd' => ['bool'],
+'XMLWriter::endDtdAttlist' => ['bool'],
+'XMLWriter::endDtdElement' => ['bool'],
+'XMLWriter::endDtdEntity' => ['bool'],
 'XMLWriter::endElement' => ['bool'],
-'XMLWriter::endPI' => ['bool'],
-'XMLWriter::flush' => ['', 'empty='=>'bool', 'xmlwriter='=>''],
+'XMLWriter::endPi' => ['bool'],
+'XMLWriter::flush' => ['string|int|false', 'empty='=>'bool'],
 'XMLWriter::fullEndElement' => ['bool'],
 'XMLWriter::openMemory' => ['bool'],
-'XMLWriter::openURI' => ['bool', 'uri'=>'string'],
-'XMLWriter::outputMemory' => ['string', 'flush='=>'bool', 'xmlwriter='=>''],
-'XMLWriter::setIndent' => ['bool', 'indent'=>'bool'],
-'XMLWriter::setIndentString' => ['bool', 'indentstring'=>'string'],
+'XMLWriter::openUri' => ['bool', 'uri'=>'string'],
+'XMLWriter::outputMemory' => ['string', 'flush='=>'bool'],
+'XMLWriter::setIndent' => ['bool', 'enable'=>'bool'],
+'XMLWriter::setIndentString' => ['bool', 'indentation'=>'string'],
 'XMLWriter::startAttribute' => ['bool', 'name'=>'string'],
-'XMLWriter::startAttributeNS' => ['bool', 'prefix'=>'string', 'name'=>'string', 'uri'=>'string'],
-'XMLWriter::startCData' => ['bool'],
+'XMLWriter::startAttributeNs' => ['bool', 'prefix'=>'string', 'name'=>'string', 'namespace'=>'?string'],
```

Comment:
I thought you said `$prefix` was nullable?

Reply:
Yes, but in the case of `startAttributeNs` it only is as of PHP 8.0: https://www.php.net/manual/en/xmlwriter.startattributens.php#refsect1-xmlwriter.startattributens-changelog. That's why I also added it to the 8.0-delta.callmap: https://github.com/vimeo/psalm/pull/6099/files#diff-5611cd00854405fe61befddd211dacd225d9c479b499ffd7e3c0c7d96eb568edR267

Reply:
And `CallMap.php` targets PHP 8.0, while delta-8.0 is used to reconstruct callmap for 7.4, delta-7.4 is used to reconstruct callmap for 7.3 and so on. So it should be nullable here.

Reply:
Ah, I misunderstood how the callmap deltas work then. I thought they were applied as upward deltas on top of `CallMap.php`, but they are applied as downward deltas. That makes sense. I fixed that now.

I probably got confused because `CallMap.php` already contained the old PHP 7.4 signatures for `xmlwriter_*` (which accept `resource`) instead of the new PHP 8.0 signatures where the `resource` type has been replaced by `XMLWriter`. I also fixed those.